mypy/checker.py

```diff
@@ -5879,15 +5879,29 @@ def find_isinstance_check_helper(
                         # considered "always right" (i.e. even if the types are not overlapping).
                         # Also note that a care must be taken to unwrap this back at read places
                         # where we use this to narrow down declared type.
-                        if node.callee.type_guard is not None:
-                            return {expr: TypeGuardedType(node.callee.type_guard)}, {}
+                        with self.msg.filter_errors(), self.local_type_map():
+                            _, real_func = self.expr_checker.check_call(
```

Comment:
The purpose of this wasn't immediately obvious. Can you add a clarifying comment?